creator_project/packages/creator-luacpp-support/reader/animation/AnimateClip.cpp

```diff
@@ -319,19 +320,19 @@ float AnimateClip::computeElapse() const
 {
     auto elapsed = _elapsed;
     auto duration = _clip->getDuration();
-    
-    // if wrap mode is pingpong or pingpongreverse, then _elapsed may be bigger than duration
-    while (elapsed >= duration)
-        elapsed = elapsed - duration;
-    
+
+    // as the time goes, _elapsed will be bigger than duration when _needStop = false
+    elapsed  = elapsed - static_cast<int>(elapsed / duration) * duration;
```

Comment:
Yep, the code is better, but how about
```c++
elapsed  %= duration;
```
?

Reply:
Both are float number, I can use the `fmodf` function